src/SessionManager.php

```diff
@@ -45,16 +65,25 @@ class SessionManager extends AbstractManager
      * @param  Storage\StorageInterface|null         $storage
      * @param  SaveHandler\SaveHandlerInterface|null $saveHandler
      * @param  array                                 $validators
+     * @param  array                                 $options
      * @throws Exception\RuntimeException
      */
     public function __construct(
         Config\ConfigInterface $config = null,
         Storage\StorageInterface $storage = null,
         SaveHandler\SaveHandlerInterface $saveHandler = null,
-        array $validators = []
+        array $validators = [],
+        array $options = []
     ) {
         parent::__construct($config, $storage, $saveHandler, $validators);
         register_shutdown_function([$this, 'writeClose']);
+
+        $this->setOptions($options);
+        if ($this->options['attach_default_validators']) {
+            $this->validators = array_merge($this->defaultValidators, $validators);
+        }
```

Comment:
The above logic should optimally occur *before* the call to `parent::__construct`; I'll make that change on merge.